src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -71,20 +70,29 @@ private enum Flags : byte
         /// the parameter can be null.
         /// </remarks>
         protected SourcePropertySymbolBase(
-           SourceMemberContainerTypeSymbol containingType,
-           Binder? binder,
-           CSharpSyntaxNode syntax,
-           RefKind refKind,
-           string name,
-           Location location,
-           DiagnosticBag diagnostics)
+            SourceMemberContainerTypeSymbol containingType,
+            Binder? binder,
+            CSharpSyntaxNode syntax,
+            CSharpSyntaxNode? getSyntax,
+            CSharpSyntaxNode? setSyntax,
+            ArrowExpressionClauseSyntax? arrowExpression,
+            ExplicitInterfaceSpecifierSyntax? interfaceSpecifier,
+            DeclarationModifiers modifiers,
+            bool isIndexer,
+            bool hasInitializer,
+            bool isAutoProperty,
+            bool hasAccessorList,
+            bool isInitOnly,
+            RefKind refKind,
+            string name,
+            Location location,
+            TypeWithAnnotations typeOpt,
+            ImmutableArray<ParameterSymbol> parametersOpt,
```

Comment:
>parametersOpt [](start = 44, length = 13)

This could either be empty, or default. Consider passing a boolean "hasParameters" instead.

Reply:
`IsDefault` captures that.

---
In reply to: [446545495](https://github.com/dotnet/roslyn/pull/45336#discussion_r446545495) [](ancestors = 446545495)

Reply:
> IsDefault captures that.

Why give an illusion that you can pass a non-empty array?

---
In reply to: [446549153](https://github.com/dotnet/roslyn/pull/45336#discussion_r446549153) [](ancestors = 446549153,446545495)

Reply:
We don't often allow a `default` value for `Parameters`.

---
In reply to: [446555527](https://github.com/dotnet/roslyn/pull/45336#discussion_r446555527) [](ancestors = 446555527,446549153,446545495)

Reply:
> We don't often allow a default value for Parameters.

Sorry, I am not following. Let me rephrase the question. Why use an array type to communicate a boolean state? Things simply wouldn't work properly if one passes an array with actual parameter symbols in it. No one makes sure the symbols would be linked properly. So, why give consumer an illusion that the constructor can properly deal with the situation?

---
In reply to: [446569515](https://github.com/dotnet/roslyn/pull/45336#discussion_r446569515) [](ancestors = 446569515,446555527,446549153,446545495)

Reply:
Makes sense, thanks.

---
In reply to: [446582904](https://github.com/dotnet/roslyn/pull/45336#discussion_r446582904) [](ancestors = 446582904,446569515,446555527,446549153,446545495)